worlds/pokemon_rb/rock_tunnel.py

```diff
@@ -1,35 +1,70 @@
+    [28, 29, 29, 29, 30, 78, 81, 82, 77, 78, 28, 29, 30, 28, 29, 29, 29, 29, 29, 30],
+]
+layout2F = [
+    [23,  2, 32, 34, 32, 34, 32, 34, 32, 34, 32, 34, 32, 34, 32, 34, 32, 34, 32, 34],
+    [31, 62,  1, 23,  1,  1, 23,  1,  1,  1,  1,  1, 23, 62,  1,  1,  1,  1,  1,  2],
+    [23,  1,  1, 31,  1,  1, 31,  1,  1,  1,  1,  1, 31,  1,  1,  1,  1,  1,  1, 23],
+    [31,  1,  1, 23,  1,  1, 23,  1,  1, 23,  1,  1, 23,  1,  1, 23, 23,  1,  1, 31],
+    [23,  1,  1, 31,  1,  1, 31,  1,  1, 31,  2,  2, 31,  1,  1, 31, 31,  1,  1, 23],
+    [31,  1,  1,  1,  1,  1, 23,  1,  1,  1,  1, 62, 23,  1,  1,  1,  1,  1,  1, 31],
+    [23,  1,  1,  1,  1,  1, 31,  1,  1,  1,  1,  1, 31,  1,  1,  1,  1,  1,  1, 23],
+    [31,  1,  1, 23,  1,  1,  1,  1,  1, 23, 32, 34, 32, 34, 32, 34,  1,  1,  1, 31],
+    [23,  1,  1, 31,  1,  1,  1,  1,  1, 31,  1,  1,  1,  1,  1,  1,  1,  1,  1, 23],
+    [31,  1,  1, 23,  1,  1,  2,  1,  1, 23,  1,  1,  1,  1,  1,  1,  1,  1,  1, 31],
+    [23,  1,  1, 31,  1,  1,  2,  1,  1, 31,  1,  1,  1, 32, 34, 32, 34, 32, 34, 23],
+    [31,  2,  2,  2,  1,  1, 32, 34, 32, 34,  1,  1,  1, 23,  1,  1,  1,  1,  1, 31],
+    [23,  1,  1,  1,  1,  1, 23,  1,  1,  1,  1,  1,  1, 31,  1,  1, 62,  1,  1, 23],
+    [31,  1,  1,  1,  1,  1, 31,  1,  1,  1,  1,  1,  1, 23,  1,  1,  1,  1,  1, 31],
+    [23, 32, 34, 32, 34, 32, 34,  1,  1, 32, 34, 32, 34, 31,  1,  1,  1,  1,  1, 23],
+    [31,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1, 31],
+    [ 2,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1, 23],
+    [32, 34, 32, 34, 32, 34, 32, 34, 32, 34, 32, 34, 32, 34, 32, 34, 32, 34,  2, 31]
+]
+
 disallowed1F = [[2, 2], [3, 2], [1, 8], [2, 8], [7, 7], [8, 7], [10, 4], [11, 4], [11, 12],
               [11, 13], [16, 10], [17, 10], [18, 10], [16, 12], [17, 12], [18, 12]]
 disallowed2F = [[16, 2], [17, 2], [18, 2], [15, 5], [15, 6], [10, 10], [11, 10], [12, 10], [7, 14], [8, 14], [1, 15],
                 [13, 15], [13, 16], [1, 12], [1, 10], [3, 5], [3, 6], [5, 6], [5, 7], [5, 8], [1, 2], [1, 3], [1, 4],
                 [11, 1]]
 
 
-def randomize_rock_tunnel(data, random):
+def randomize_rock_tunnel(patch: typing.Union["PokemonBlueProcedurePatch", "PokemonRedProcedurePatch"], random: random.Random):
+    import copy
 
     seed = random.randint(0, 999999999999999999)
     random.seed(seed)
 
-    map1f = []
-    map2f = []
-
-    address = rom_addresses["Map_Rock_Tunnel1F"]
-    for y in range(0, 18):
-        row = []
-        for x in range(0, 20):
-            row.append(data[address])
-            address += 1
-        map1f.append(row)
-
-    address = rom_addresses["Map_Rock_TunnelB1F"]
-    for y in range(0, 18):
-        row = []
-        for x in range(0, 20):
-            row.append(data[address])
-            address += 1
-        map2f.append(row)
+    map1f = copy.deepcopy(layout1F)
+    map2f = copy.deepcopy(layout2F)
```

Comment:
afaik deepcopy is expensive and since we know this data is just list[list[int]] this should work fine
```suggestion
    map1f = [row.copy() for row in layout1F]
    map2f = [row.copy() for row in layout2F]
```

Reply:
I'm not super surprised that this is faster, but I am surprised that it's an order of magnitude faster.